pkg/apis/traceflow/v1/types.go

```diff
@@ -0,0 +1,158 @@
+// Copyright 2020 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type DropReason string
+
+type Phase string
+
+const (
+	Initial Phase = "Initial"
+	Running Phase = "Running"
+	Success Phase = "Success"
+	Timeout Phase = "Timeout"
+	Error   Phase = "Error"
+)
+
+type Component string
+
+const (
+	SpoofGuard    Component = "SpoofGuard"
+	LB            Component = "LB"
+	Routing       Component = "Routing"
+	NetworkPolicy Component = "NetworkPolicy"
+	Forwarding    Component = "Forwarding"
+)
+
+type Action string
+
+const (
+	Delivered Action = "Delivered"
+	Received  Action = "Received"
+	Forwarded Action = "Forwarded"
+	Dropped   Action = "Dropped"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type Traceflow struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   `json:"spec"`
+	Status `json:"status"`
+}
+
+// Spec describes the spec of the traceflow.
+type Spec struct {
+	Source      `json:"source"`
+	Destination `json:"destination"`
+	Packet      `json:"packet"`
+}
+
+// Source describes the source spec of the traceflow.
+type Source struct {
+	Namespace string `json:"namespace,omitempty"` // Namespace is the source namespace.
```

Comment:
I know there are some comments are appended to the lines of code which causes openapi generates wrong document.
Please don't add more and just add comments in the normal style:
```
// Namespace is the source namespace.
Namespace string `json:"namespace,omitempty"`
```

Reply:
Updated.